docs/reference/setup/install/getting-started-tls-docker.asciidoc

```diff
@@ -0,0 +1,171 @@
+    working_dir: /usr/share/elasticsearch
+    volumes: ['.:/usr/share/elasticsearch/config/x-pack/certificates']
+----
+
+<1> The new node certificates and CA certificate+key are placed under the local directory `certs`.
+endif::[]
+
+[[getting-starter-tls-create-docker-compose]]
+`docker-compose.yml`:
+ifeval::["{release-state}"=="unreleased"]
+
+WARNING: Version {version} of Elasticsearch has not yet been released, so a
+`docker-compose.yml` is not available for this version.
+
+endif::[]
+
+ifeval::["{release-state}"!="unreleased"]
+["source","yaml",subs="attributes"]
+----
+version: '2.2'
+services:
+  es01:
+    container_name: es01
+    image: docker.elastic.co/elasticsearch/elasticsearch-platinum:{version}
+    environment:
+      - node.name=es01
+      - ELASTIC_PASSWORD=$ELASTIC_PASSWORD <1>
+      - "ES_JAVA_OPTS=-Xms512m -Xmx512m"
+      - xpack.security.http.ssl.enabled=true
+      - xpack.security.transport.ssl.enabled=true
+      - xpack.security.transport.ssl.verification_mode=certificate <2>
+      - xpack.ssl.certificate_authorities=$CERTS_DIR/ca/ca.crt
+      - xpack.ssl.certificate=$CERTS_DIR/es01/es01.crt
+      - xpack.ssl.key=$CERTS_DIR/es01/es01.key
+    volumes: ['esdata_01:/usr/share/elasticsearch/data', './certs:$CERTS_DIR']
+    ports:
+      - 9200:9200
+    healthcheck:
+      test: curl --cacert $CERTS_DIR/ca/ca.crt -s https://localhost:9200 >/dev/null; if [[ $$? == 52 ]]; then echo 0; else echo 1; fi
+      interval: 30s
+      timeout: 10s
+      retries: 5
+  es02:
+    container_name: es02
+    image: docker.elastic.co/elasticsearch/elasticsearch-platinum:{version}
+    environment:
+      - node.name=es02
+      - ELASTIC_PASSWORD=$ELASTIC_PASSWORD
+      - discovery.zen.ping.unicast.hosts=es01
+      - "ES_JAVA_OPTS=-Xms512m -Xmx512m"
+      - xpack.security.http.ssl.enabled=true
+      - xpack.security.transport.ssl.enabled=true
+      - xpack.security.transport.ssl.verification_mode=certificate
+      - xpack.ssl.certificate_authorities=$CERTS_DIR/ca/ca.crt
+      - xpack.ssl.certificate=$CERTS_DIR/es02/es02.crt
+      - xpack.ssl.key=$CERTS_DIR/es02/es02.key
+    volumes: ['esdata_02:/usr/share/elasticsearch/data', './certs:$CERTS_DIR']
+  wait_until_ready:
+    image: docker.elastic.co/elasticsearch/elasticsearch-platinum:{version}
+    command: /usr/bin/true
+    depends_on: {"es01": {"condition": "service_healthy"}}
+volumes: {"esdata_01": {"driver": "local"}, "esdata_02": {"driver": "local"}}
+----
+
+<1> Bootstrap `elastic` with the password defined in `.env`. See {xpack-ref}/setting-up-authentication.html#bootstrap-elastic-passwords[the Elastic Boostrap Password].
+<2> Disable verification of authenticity for inter-node communication. Allows creating self-signed certificates without having to pin specific internal IP addresses.
+endif::[]
+
+==== Run the example
+. Generate the certificates (only needed once):
++
+--
+["source","sh"]
+----
+docker-compose -f create-certs.yml up
+----
```

Comment:
Is it worth making the node containers "depend" on this one instead? Will that work? Is it too fiddly?

Reply:
As discussed this is very tricky, because the create-certs docker container will eventually exit and it's hard to create a decent healthcheck in this case.